x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/task/ReindexDataStreamClient.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.migrate.task;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.support.AbstractClient;
+import org.elasticsearch.xpack.core.ClientHelper;
+
+import java.util.Map;
+
+public class ReindexDataStreamClient extends AbstractClient {
```

Comment:
This one isn't really specific to reindex, since it doesn't do much other than store the client and headers and use it to execute the request. Does it make sense to name it something more generic? Or to put it somewhere more generic?

Reply:
Yeah good point. We'll probably wind up using the same code for other parts of the upgrade. I couldn't think of a great name, so I renamed it to the more generic `ExecuteWithHeadersClient`.